### ISSUE TITLE
Speed up build/compilation

### ISSUE THREAD

**Issue by tizmagik** [NYTimes/kyt#216]
The current build time is extremely slow and will grow as the codebase does. I'm not as concerned with the production build time as I am about the incremental builds for hot module reloading since that's what impacts DX most, but I'd expect both to be improved with some of the following solutions.

There's a few things we can try:

- [HappyPack](https://github.com/amireh/happypack)
- [Webpack DLL Plugin](https://github.com/webpack/webpack/tree/master/examples/dll)
- Use [babel-preset-node6](https://github.com/salakar/babel-preset-node6) for server bundle

Some helpful reads/links:

- Webpack issue with a lot of good thoughts: https://github.com/webpack/webpack/issues/1574#issuecomment-189580065
- [Blog post on using DLL plugin](https://medium.com/@soederpop/webpack-plugins-been-we-been-keepin-on-the-dll-cdfdd6cb8cd7#.8k0bmkjbm)
- [SlideShare on making webpack builds 10x faster](http://www.slideshare.net/trueter/how-to-make-your-webpack-builds-10x-faster)

I'm sure there's a handful of other levers we could pull, please feel free to add more info! 

I'm going to start by implementing these in our internal project first and then seeing how we can cleanly abstract this upstream into `kyt`. But individual PRs are welcome!

**Comment by delambo** [comment]
For scripts, I think a lot of the build time is in the source map generation. By default we use the webpack `devtool:source-map` which was yielding ~3s rebuilds:

```
webpack built d21695c5a054775f4ea1 in 2912ms
...
webpack built 002e7877be1aecb09a84 in 2459ms
...
webpack built 7e30df74a806e6d12210 in 2685ms
```

When I switched it to `devtool:cheap-eval-source-map` the rebuild times were around a second:
```
webpack built fb24531f5c2a8d983afe in 810ms
...
webpack built deb8470a92ad180eb191 in 861ms
...
webpack built 2e948108620df32decd8 in 941ms
```

**Comment by tizmagik** [comment]
Yea that's why I suggested using cheap-module-eval-source-map in #219 -- it has the best speed to source map (line numbers etc) trade-off and should be used in development. For production builds `source-map` is preferred to generate an external map file and we're not so concerned with build speed there.

**Comment by kjg531** [comment]
has anyone successful implemented Happypack within KYT?

**Comment by jamieholliday** [comment]
I have tried setting up DLL plugin using kyt.conf.js One of the main issue I am finding is where to put the generated dll file. This needs to be accessible to be loaded by server/template.js so my usual approach to this would be to put in the build dir. However this dir gets cleaned out when running 'kyt dev' command. I did managed to hack it around a bit to test and noticed faster rebuild times on a test app from ~5s to ~2.5s so I'd be quite keen to figure out if this is possible to set this up using kyt

**Comment by delambo** [comment]
Hey, @jamieholliday - what are you using the DLL plugin for? I was going to investigate using it for producing a vendor bundle in #315.

**Comment by jamieholliday** [comment]
Yes exactly that producing a vendor bundle. I will keep an eye on #315 then. Thanks

**Comment by tizmagik** [comment]
Just came across [autodll-webpack-plugin](https://github.com/asfktz/autodll-webpack-plugin) which looks really nice and should significantly speed up compilation. I think this requires us to be using the `HtmlWebpackPlugin`, which might be a good idea in general.
